segmentation/utils/synthetic_spatial_datagen.py

```diff
@@ -0,0 +1,204 @@
+    first_row = np.concatenate(((random_aa + random_aa.T) / 2, random_ab, random_ac), axis=1)
+    second_row = np.concatenate((random_ab.T, (random_bb + random_bb.T) / 2, random_bc), axis=1)
+    third_row = np.concatenate((random_ac.T, random_bc.T, (random_cc + random_cc.T) / 2), axis=1)
+
+    # then concatenate them together
+    dist_mat = np.concatenate((first_row, second_row, third_row), axis=0)
+
+    # assert that the created submatrix is symmetric
+    assert np.allclose(dist_mat, dist_mat.T, rtol=1e-05, atol=1e-08)
+
+    return dist_mat
+
+
+def get_random_centroid_centers(size_img=(1024, 1024), num_A=100, num_B=100, num_C=100, distr_A=None, distr_B=None, distr_C=None, seed=None):
+    """
+    This function generates random centroid centers such that those of type A will have centers
+    closer on average to those of type B than those of type C
+
+    We will use a multivariate Gaussian distribution for A, B, and C type cells to generate their respective centers.
+
+    Returns the set of points associated with the centroids of cells of types A, B, and C.
+
+    Args:
+        size_img: a tuple indicating the size of the image. Default 1024 x 1024
+        num_A: the number of A centroids to generate. Default 100.
+        num_B: the number of B centroids to generate. Default 100.
+        num_C: the number of C centroids to generate. Default 100.
+
+        distr_A: a dict indicating the parameters of the multivariate normal distribution to generate A cell centroid.
+            If None, use predefined parameters.
+        distr_B: similar to distr_A
+        distr_C: similar to distr_C
+        seed: whether to fix the random seed or not. Useful for testing.
+            Should be a specified integer value. Default None.
+    """
+
+    # extract the height and width
+    height = size_img[0]
+    width = size_img[1]
+
+    # set the distribution parameters, use the mean scaling factor and default covariance if not specified
+    if not distr_A:
+        a_mean = (height * A_CENTROID_FACTOR, width * A_CENTROID_FACTOR)
+        a_cov = A_CENTROID_COV
+    else:
+        mean_a = distr_A['mean']
+        var_a = distr_A['cov']
+
+    if not distr_B:
+        b_mean = (height * B_CENTROID_FACTOR, width * B_CENTROID_FACTOR)
+        b_cov = B_CENTROID_COV
+    else:
+        mean_b = distr_B['mean']
+        var_b = distr_B['cov']
+
+    if not distr_C:
+        c_mean = (height * C_CENTROID_FACTOR, width * C_CENTROID_FACTOR)
+        c_cov = C_CENTROID_COV
+    else:
+        mean_ac = distr_C['mean']
+        var_ac = distr_C['var']
+
+    # if specified, set the random seed
+    if seed:
+        np.random.seed(seed)
+
+    # use the multivariate_normal distribution to generate the points
+    a_points = np.random.multivariate_normal(a_mean, a_cov, num_A)
+    b_points = np.random.multivariate_normal(b_mean, b_cov, num_B)
+    c_points = np.random.multivariate_normal(c_mean, c_cov, num_C)
+
+    return a_points, b_points, c_points
```

Comment:
doesn't really matter but it's probably better to return these as a tuple or a dictionary.  i'd go tuple personally, but i have nothin against dictionaries